torch/_inductor/autotune_process.py

```diff
@@ -97,18 +84,59 @@ def initialize(self) -> None:
             atexit.register(lambda: self.terminate())
 
         # wait for the initialization to be done
-        request_queue.put(Ping())
-        resp = response_queue.get()
+        self.put(Ping())
+        resp = self.get()
         assert isinstance(resp, Pong)
 
+    def put(self, obj: Any) -> None:
+        """
+        Push a work item to the child process.
+        """
+        assert self.process is not None
+        assert self.process.stdin is not None
+        pickle.dump(obj, self.process.stdin)
+        self.process.stdin.flush()
+
+    def get(self) -> Any:
+        """
+        Get a response from the child process.
+        """
+        assert self.process is not None
+        assert self.process.stdout is not None
+        try:
+            return pickle.load(self.process.stdout)
+        except EOFError:
+            # Child crashed; recreate it
+            self.close()
+            self.initialize()
+            raise
```

Comment:
for my own understanding, is there a reason why we initialize before re-throwing ?

Reply:
@eellison: I'm sure we could do it elsewhere if it makes more sense. It made sense to me to keep the management of the sub-process wholly contained within this class (i.e., the TuningProcess object makes sure its sub-process is always running). This is the most immediate point where we know the sub-process exited, so I just restart it immediately. Another approach is to re-initialize, if necessary, in the put() method. Or get even more elaborate and figure out how to reinitialize asynchronously so we don't block on the sub-process creation?

Reply:
Just FYI, remove `self.initialize` here is fine since we call `self.initialize` in benchmark_in_sub_proc` anyways. Removing the call here makes the sub-process recreation lazy.

Reply:
Sure, we can do lazy